Gems/OpenXRVk/Code/Source/XRCameraMovementComponent.h

```diff
@@ -0,0 +1,116 @@
+
+        /*
+        * Specifies the services that this component requires.
+        * The system activates the required services before it activates this component.
+        * It also deactivates the required services after it deactivates this component.
+        * If a required service is missing before this component is activated, the system
+        * returns an error and does not activate this component.
+        */
+        static void GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required);
+
+        /*
+        * Specifies the services that this component depends on, but does not require.
+        * The system activates the dependent services before it activates this component.
+        * It also deactivates the dependent services after it deactivates this component.
+        * If a dependent service is missing before this component is activated, the system
+        * does not return an error and still activates this component.
+        */
+        static void GetDependentServices(AZ::ComponentDescriptor::DependencyArrayType& dependent);
+
+    protected:
+        /*
+        * Puts this component into an active state.
+        * The system calls this function once during activation of each entity that owns the
+        * component. You must override this function. The system calls a component's Activate()
+        * function only if all services and components that the component depends on are present
+        * and active.
+        */
+        void Activate() override;
+
+        /*
+        * Deactivates this component.
+        * The system calls this function when the owning entity is being deactivated. You must
+        * override this function. As a best practice, ensure that this function returns this component
+        * to a minimal footprint. The order of deactivation is the reverse of activation, so your
+        * component is deactivated before the components it depends on.
+        *
+        * The system always calls a component's Deactivate() function before destroying the component.
+        * However, deactivation is not always followed by the destruction of the component. An entity and
+        * its components can be deactivated and reactivated without being destroyed. Ensure that your
+        * Deactivate() implementation can handle this scenario.
+        */
+        void Deactivate() override;
+
+        // AZ::TickBus
+        void OnTick(float deltaTime, AZ::ScriptTimePoint timePoint) override;
+
+        // AzFramework::InputChannelEventListener
+        bool OnInputChannelEventFiltered(const AzFramework::InputChannel& inputChannel) override;
+
+    private:
+        void OnXRControllerEvent(const AzFramework::InputChannel& inputChannel);
+        static AZ::Transform GetCameraTransformFromCurrentView();
+
+        // Transient data...
+        AZ::Vector3 m_movement{ AZ::Vector3::CreateZero() };    // Only use the X/Y components of this vector
+        float m_heightZ{ 0.f };
```

Comment:
We use following convention everywhere in XR gems so maybe good to stick with this everywhere. 
float m_heightZ = 0.f;

Reply:
Sorry, it's my habit to use curly initializers nearly everywhere.  I'll update it.